fs-storage/src/folder_storage.rs

```diff
@@ -0,0 +1,784 @@
+                )
+            })?
+            .parse::<K>()
+            .map_err(|_| {
+                ArklibError::Storage(
+                    self.label.clone(),
+                    "Failed to parse key from filename".to_owned(),
+                )
+            })
+    }
+}
+
+impl<K, V> BaseStorage<K, V> for FolderStorage<K, V>
+where
+    K: Ord
+        + Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::str::FromStr
+        + std::fmt::Display,
+    V: Clone + serde::Serialize + serde::de::DeserializeOwned + Monoid<V>,
+{
+    /// Set a key-value pair in the internal mapping
+    fn set(&mut self, key: K, value: V) {
+        self.data.insert(key.clone(), value);
+        self.deleted_keys.remove(&key);
+        self.modified.insert(key, SystemTime::now());
+    }
+
+    /// Remove an entry from the internal mapping given a key
+    fn remove(&mut self, id: &K) -> Result<()> {
+        match self.data.remove(id) {
+            Some(_) => {
+                self.deleted_keys.insert(id.clone());
+                Ok(())
+            }
+            None => Err(ArklibError::Storage(
+                self.label.clone(),
+                "Key not found".to_owned(),
+            )),
+        }
+    }
+
+    /// Compare the timestamp of the storage files
+    /// with the timestamps of the in-memory storage and the last written
+    /// to time to determine if either of the two requires syncing.
+    fn sync_status(&mut self) -> Result<SyncStatus> {
+        let mut ram_newer = !self.deleted_keys.is_empty();
+        let mut disk_newer = false;
+
+        for key in self.data.keys() {
+            let file_path = self.path.join(format!("{}.json", key));
+            let ram_timestamp = self
+                .modified
+                .get(key)
+                .expect("Data entry key should have ram timestamp");
+
+            if let Ok(metadata) = fs::metadata(&file_path) {
+                if let Ok(disk_timestamp) = metadata.modified() {
+                    match ram_timestamp.cmp(&disk_timestamp) {
+                        std::cmp::Ordering::Greater => {
+                            ram_newer = true;
+                            log::debug!(
+                                "RAM newer: file {} is newer in RAM",
+                                file_path.display()
+                            );
+                        }
+                        std::cmp::Ordering::Less => {
+                            disk_newer = true;
+                            log::debug!(
+                                "Disk newer: file {} is newer on disk, ram: {}, disk: {}",
+                                file_path.display(),
+                                ram_timestamp.elapsed().unwrap().as_secs(),
+                                disk_timestamp.elapsed().unwrap().as_secs()
+                            );
+                        }
+                        std::cmp::Ordering::Equal => {}
```

Comment:
Could we use `>` and `<` instead? I feel it's more intuitive.

Reply:
I tried it, but Clippy suggests using `cmp` & `match` instead of this comparison_chain.

```bash
warning: `if` chain can be rewritten with `match`
   --> fs-storage/src/folder_storage.rs:268:21
    |
268 | /                     if ram_timestamp > &disk_timestamp {
269 | |                         ram_newer = true;
270 | |                         log::debug!(
271 | |                             "RAM newer: file {} is newer in RAM",
...   |
281 | |                         );
282 | |                     }
    | |_____________________^
    |
    = help: consider rewriting the `if` chain to use `cmp` and `match`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
    = note: `#[warn(clippy::comparison_chain)]` on by default

```

Reply:
Ok then it's fine. Not a big deal
Thank you

Reply:
Interesting, maybe if it was one `if` after another